### PR TITLE
Utils transforms refactoring

### PR DESCRIPTION
**Pull request by mpreisler**
I am doing some refactoring and cleanup. Opened this pull request to discuss some of the changes so that I don't waste effort before I rip stuff out. Only we are using the build system so let's rip out the old stuff so that we aren't dragging it forward forever.

Questions for @iankko, @redhatrises, @shawndwells, @jan-cerny, @ybznek:
- ~~Can I rip out the test profile? I have a suspicion that nobody is using it and it complicates the build process.~~ DONE
- ~~Can I rip out `make eval-test`, `make eval-common`? Is anybody using that? I think we should use the build system to build stuff, not to test it.~~ DONE
- ~~The build output is spammed with comments. What do you think about hiding at least some of those using `@#` instead of just `#` as prefix?~~ Let's deal with this separately.
- ~~Build all remediations from templates every time we build. Don't commit generated remediation fixes in git. Yes or no? This will make the build take slightly more time but compared to generating guides it's miniscule.~~ We agree that we want this but @ybznek will do it in a separate PR. I will not handle this in this PR.

My long term goal is to clean up and simplify the build so that I can move it to cmake in the future which will speed rebuilds dramatically. Because of proper target dependency mappings we will gain by that.

### CODE REVIEW

**Comment by redhatrises** [comment]
> - Can I rip out the test profile? I have a suspicion that nobody is using it and it complicates the build process.

I am fine with this as I also do not think that anybody is using it. I think we should build a test suite instead using something like pytest.

> - Can I rip out `make eval-test`, `make eval-common`? Is anybody using that? I think we should use the build system to build stuff, not to test it.

I would be fine removing these targets as well as I also doubt that anybody is using it.

> - The build output is spammed with comments. What do you think about hiding at least some of those using `@#` instead of just `#` as prefix?

I don't think that we should be showing comments at all. Side note: I also thought that you can remove the tabs, and comments would not show as well?

> - Build all remediations from templates every time we build. Don't commit generated remediation fixes in git. Yes or no? This will make the build take slightly more time but compared to generating guides it's miniscule.

+1. However being able to take a script in and out of templated form should be taken into account because some scripts that are templated may be customized due to use cases so that they are no longer templated. In the same vain, some scripts that are not templated may become templated due to use cases. 

> My long term goal is to clean up and simplify the build so that I can move it to cmake in the future which will speed rebuilds dramatically. Because of proper target dependency mappings we will gain by that.

+1... It would be really awesome for `oscap` to use cmake as well... ;)

**Comment by iankko** [comment]
Hi Martin,

thanks for checking.
>  Can I rip out the test profile? I have a suspicion that nobody is using it and it complicates the build process.

To my knowledge the ```test``` profile was there just to quickly test the content is working after RPM package build (some lightweight profile the scan not to take that long time). But since within the process of making the SSG upstream release we test the work of the content on couple of selected profiles (I am using scap-workbench tool for that), I would say it's OK to remove the test profile.

> Can I rip out make eval-test, make eval-common? Is anybody using that? I think we should use the build system to build stuff, not to test it.

Assuming the issue here aren't the ```make eval-test``` and ```make eval-common``` targets themselves, but the fact for the Makefiles they are available these making the difference from other Makefiles (products), when these aren't available, right? (and since we would to unify the form of Makefiles to be as much aligned for different products as possible this isn't currently possible just because some Makefiles have these targets, while others not).

If the above is the case, what to drop existing ```make eval-test``` and ```make eval-common``` targets (as they exist right now). And instead of that introduce one ```make test``` target. This would do the following:
* scan the existing benchmark for available profiles (thus would depend on ```make content``` for the product),
* and would call ```make eval profile-id``` for each profile present in that benchmark.

This way we would have unified Makefile form, but on the other hand wouldn't lose some extra way how to automatically check the content via oscap run. The ``make test``` target wouldn't be run by default when making RPM. But IMHO could be run within ```scap-security-guide``` Jenkins CI job:
  https://jenkins.open-scap.org/job/scap-security-guide/

(not the ```scap-security-guide-pull-requests``` one -- the PR check not to take that long. But IMHO it could run within ```scap-security-guide``` Jenkins CI, the purpose of which right now is to check if content builds correctly. On that job IMHO it wouldn't mind if it would take a bit longer (till all profiles would be evaluated). And we could ensure all profiles have sane content (meaning attempt to load the benchmark and scan the benchmark in the scanner works properly). It would be sufficient if that test would fail till next day (so in the case we regressed [some profile is unable to load e.g. due to non-existing selectors used in the profile]), we could fix the issue promptly. What do you think?

> The build output is spammed with comments. What do you think about hiding at least some of those using @# instead of just # as prefix?

No objections against this from me. Please go ahead.

>    Build all remediations from templates every time we build. Don't commit generated remediation fixes in git. Yes or no? This will make the build take slightly more time but compared to generating guides it's miniscule.

IMHO the build time wouldn't change that much when this is applied. But like Gabe mentioned, some remediations have been altered since being originally created from template (I tend to remote the "Generated by ..." comment for such case) due some need. On the other hand, some have been templated later (even if not created via template originally). 

In short, I have no objections against building remediations (even OVALs in the future) from the templates under assumption / preserving the condition, the build would allow some remediations to stand outside of the templates (still to be created / modified by hand).

> My long term goal is to clean up and simplify the build so that I can move it to cmake in the future which will speed rebuilds dramatically. Because of proper target dependency mappings we will gain by that.

Right. Reasonable and long deserved Makefile layouts refactoring. I can only agree on the necessity of this. Please go ahead.

**Comment by ybznek** [comment]
I agree with 
- `Build all remediations from templates every time we build` 
- `The build output is spammed with comments. What do you think about hiding at least some of those using @# instead of just # as prefix?` 

but don't have opinion for other answers.

**Comment by mpreisler** [comment]
Ready for review & merge.

**Comment by redhatrises** [comment]
Initial look through looks good to me. Are you going to take of the last two items in a separate PR?

**Comment by mpreisler** [comment]
> Initial look through looks good to me. Are you going to take of the last two items in a separate PR?

remediation building will be done by @ybznek . The comments I hope to ignore for now and then fix them once and for all with the transition to cmake.

**Comment by ybznek** [comment]
LGTM. I was afraid due to python changes, but on jenkins it passed with python2 & python3 and I haven't found any problem with generated rhel5 content.
